components/funder/src/mutual_credit/incoming.rs

```diff
@@ -262,7 +262,8 @@ fn process_response_send_funds(
         .ok_or(ProcessOperationError::RequestDoesNotExist)?
         .clone();
 
-    let dest_public_key = pending_transaction.route.public_keys.last().unwrap();
+    // Note the key order is reversed from our perspective
+    let dest_public_key = &mutual_credit.state().idents.remote_public_key;
```

Comment:
@pzmarzly : I overlooked this problem when I wrote the instructions for making the modifications. Sorry about that!

The proposed code is incorrect. For example, consider a the network topology:
`B -- C -- D -- E`, where B sends a request along this route to E, and then E sends back a response all the way back to B. The response is signed by E.

When C gets the request, he is given the route `D -- E` and keeps it. Later, when the response goes back from D to C, C needs to check the signature. If C uses `remote_public_key`, he will check the signature using `D`'s public key, which is incorrect. The right thing to do is to check it using the last public key on the route he has, which is `E`.

From what I understand an incoming response can never have an empty route. If it had an empty route, it means that we are the destination of the request, which means we sent the response. It is not possible to have it as incoming response. So I think you can leave the original line. (Unwraping the last item of the route).

Reply:
> From what I understand an incoming response can never have an empty route

Well, these 3 tests panic on this `unwrap`:

```text
    handler::tests::pair_basic::test_handler_pair_basic
    tests::tests::test_funder_basic
    tests::tests::test_funder_forward_payment
```

but `if-else` approach similar to `outgoing.rs` makes them work.